framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
> Yes, I've already change `supprotedQuals` a non-final field.

I changed my mind. To using lazy initialisation on `supportedQuals`, there is a tradeoff:

1) makes `supportedQuals` a non-final field, which means we lose the reference immutability of `supportedQuals`, then we let `getSupportedQualifiers()` assign `supportedQuals` to an immutable set at each time it gets called:

``` java
supportedQuals = Collections.unmodifiableSet(supportedQuals);
return supportedQuals;
```

2) keep `supportedQuals` a final field, but now `supportedQuals` is a mutable set, thus `getSupportedQualifiers()` should ensure that it always return an immutable view of `supportedQuals` to the outside of `AnnotatedTypeFactory`:
```java
return Collections.unmodifiableSet(supportedQuals);
```

I prefer the latter one, since although we loose the `supportedQuals` to a mutable set, we still make sure it always refer to one same set.

In former decision, we can reset `supportedQuals` refer to a mutable set, and do whatever we want to `supportedQuals`.